diesel/src/query_dsl/mod.rs

```diff
@@ -836,8 +836,8 @@ pub trait QueryDsl: Sized {
 
     /// Sets the `group by` clause of a query.
     ///
-    /// **Note:** Queries having a `group by` clause require a custom select clause.
-    /// Use `QueryDsl::select()` to specify one
+    /// **Note:** Queries having a `group by` clause require a custom select clause _after_ it.
+    /// Use `QueryDsl::select()` to specify one.
```

Comment:
```suggestion
    /// **Note:** Queries having a `group by` clause require a custom select clause.
    /// Use `QueryDsl::select()` to specify one. `select` needs to be before `group_by`
    /// if there are group by clause within select clause.
```

Reply:
Oops, I accidently typed the wrong thing.

```suggestion
    /// **Note:** Queries having a `group by` clause require a custom select clause.
    /// Use `QueryDsl::select()` to specify one. `select` needs to be after `group_by`
    /// if some fields depends on the fields within `group_by`.
```

`s/before/after` which is a typo, and the select needs to be after it if some fields is depending on the group by fields, is that correct and understandable?

Reply:
I do not feel that this is correct either. I mean a select clause does not consist of fields (`users.name`), but of expressions  (e.g. `avg(users.age) + users.id` to have something that does not make much sense). Also as noted above this is not really clear on when you require a select clause after `group_by` and when not. Unfortunately that's something that's quite complex to get really right in one sentence. Again as mentioned already several times: A select clause must always be valid for the current `group by` clause (for what this means look up the postgres docs on interaction between group by and select). So if a select comes before `group_by` it must be valid for no group by clause as well. I do not find that very surprising as this is only the type state equivalent of the normal builder pattern.

Reply:
I guess maybe this patch is not necessary, it isn't something that can be put into words.